deploy/eck-integrations/values.yaml

```diff
@@ -0,0 +1,93 @@
+
+
+elasticsearch:
+  host: ""
+  user: ""
+  pass: ""
+
+cloudDefend:
+  enabled: false
+
+kubernetes: 
+  version: 1.51.0
+  enabled: false
+  namespace: "default"
+      
+  system:
+    metrics:
+      enabled: true
+
+  containers:
+    metrics:
+      enabled: true
+    state:
+      enabled: true
+    logs:
+      enabled: true
+  
+  objects:
+    pods:
+      metrics:
+        enabled: true
```

Comment:
I think yes that object should be removed. 
My take is that we should move at top level the metrics and logs. 
Then under metrics we should put all the objects that are relevant
Same for logs

So yes we should remove both object and control_plane

Reply:
> My take is that we should move at top level the metrics and logs.
> Then under metrics we should put all the objects that are relevant
> Same for logs

@mlunadia Any input here? I think the mental model we're expecting users to have is "what do I want to monitor" followed by "which signals do I want for those things". This is what we have in the chart now. To me this makes sense, because as a user I'm thinking about the thing that is broken first (eg. pods) and then second which data I need to find the root cause of the issue.

@gizas, your preference is the other way: "which signals do I want" followed by "for which things do I want those signals".

Reply:
Yes indeed @joshdover . I am thinking also this from the UX, that if they (or dont) want  metrics they just disable one flag. The other way the need to go and disable all metirc references

Reply:
my 2 cents, I resonate with @joshdover on the approach `"what do I want to monitor"`->`"which signals do I want for those things"`. But I hear your point as well @gizas, and we could support such setting by defining another field under `kubernetes`, e.g. `skip_metrics`, etc. that can act as an override knob for all other (sub-)metrics keys.

Reply:
+1 on the override knob. I think it's reasonable to think that users may only want logs enabled but not metrics, for example. There should be an easy way to accomplish that.

Maybe just an overall `kubernetes.metrics.enabled` flag that takes precedence when it's `false`. Same for logs (and later traces).

Reply:
> Any input here? I think the mental model we're expecting users to have is "what do I want to monitor" followed by "which signals do I want for those things". This is what we have in the chart now. To me this makes sense, because as a user I'm thinking about the thing that is broken first (eg. pods) and then second which data I need to find the root cause of the issue.

1. System that needs to be monitored: Kubernetes, nginx, redis, etc...
2. Resources in that system: Kubernetes nodes, pods, containers...
3. Signals: Logs, metrics, traces
4. Config: eg: Location> endpoint, directory, etc...

I'm in two minds on whether signals are nested into resources or the opposite but prefer it as above with an override to reduce user pain but I'm currently unsure how this could be done in yaml.

Have we considered ideas for how are we going to serve the corresponding values.yaml to users when we have 300+ integrations? A single values.yaml would be impossible to work with.

Reply:
> Have we considered ideas for how are we going to serve the corresponding values.yaml to users when we have 300+ integrations? A single values.yaml would be impossible to work with.

@mlunadia is this resolved by our discussion earlier? To summarize we said:
- Metrics integrations would be supported via hints-based autodiscovery
- Most log integrations are supported via routing container logs in ingest pipelines to more specific integrations. No Agent configuration is required. This is essentially how logs integrations work on k8s today.

I'd also add that if there are other integrations that users want to deploy on K8s, they can set up a managed agent to do that with just a regular ECK Agent resource, separate from the agent pods that are monitoring the k8s infrastructure and containers. The reason for this is that deployment mode, resource limits, and scaling of that integration have different properties than the integrations running on the main agent nodes doing k8s monitoring.

For example, if a user wants to use an AWS integration to pull data from S3, they can either create an ECK Agent CR with a Fleet enrollment token to a policy that has this integration, or they can create an [ECK Agent CR with the input configuration](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-elastic-agent-quickstart.html) copied from the Integrations UI. They can then scale and manage that agent separate from the core agents monitoring k8s.

Reply:
If we want to simplify things for users so they only have to manage the helm chart and nothing else, we could add support for additional agents that get deployed separately. We wouldn't need to include every integration in the `values.yml` but instead have users copy the config from the Integrations UI for the integrations they want OR use a Fleet-managed agent.

So we could have something like this in the values.yml which is essentially just a list of Agent CRs:

```yaml
additionalAgents:
# key is `metadata.name` for the Agent CR
myAgent:
  # contents are the same as `spec` in the Agent CR
  version: 8.11.1
  deployment: {}
  config:
    inputs:
    # copy/pasted inputs from Integrations UI
```

Reply:
with the latest iteration this is now supported, users can specify custom integrations and tight them to custom agent presets [example](https://github.com/elastic/cloud-on-k8s/pull/7356/files#diff-8b91b0bf85c2ba0d1636bacc927e5b6952590b9aacecebcdaa9f2087bdbaa2cf).